datastream-server/src/main/java/com/linkedin/datastream/server/assignment/BroadcastStrategy.java

```diff
@@ -102,10 +104,16 @@ public Map<String, Set<DatastreamTask>> assign(List<DatastreamGroup> datastreams
             .orElseGet(() -> getOrCreateDatastreamTask(reuseTasksPerDg, dg));
 
         currentAssignmentCopy.get(instance).remove(foundDatastreamTask);
+
+        // Prevent duplicate datastream task names
+        if (!uniqueDatastreamTaskNames.add(foundDatastreamTask.getDatastreamTaskName())) {
```

Comment:
If a duplicate task name is found, can you not create a new one right here. You don't have to skip continue.

Reply:
This will create a new task. By moving this to a while loop and manually incrementing the taskPos we essentially skip the duplicate by not increasing the taskPos variable and instead create a new datastream on a following iteration. I don't think the code would be very clean if we wanted to duplicate that same logic inside the conditional